internal/cli/artifact_build.go

```diff
@@ -41,6 +42,46 @@ func (c *ArtifactBuildCommand) Run(args []string) int {
 			app.UI.Output("\nCreated artifact v%d", buildResult.Push.Sequence)
 		}
 
+		// Show input variable values used in build
+		app.UI.Output("Variables used:", terminal.WithHeaderStyle())
+		headers := []string{
+			"Variable", "Value", "Type", "Source",
+		}
+
+		tbl := terminal.NewTable(headers...)
+		// sort alphabetically for joy
+		inputVars := make([]string, 0, len(buildResult.Build.VariableRefs))
+		for iv := range buildResult.Build.VariableRefs {
+			inputVars = append(inputVars, iv)
+		}
+		sort.Strings(inputVars)
+		for _, iv := range inputVars {
+			// We add a line break in the value here because the Table word wrap
+			// alone can't accomodate the column headers to a long value
+			val := buildResult.Build.VariableRefs[iv].Value
+			if len(val) > 45 {
+				for i := range val {
+					// line break every 45 characters
+					if i%46 == 0 && i != 0 {
+						val = val[:i] + "\n" + val[i:]
+					}
+				}
+			}
+			columns := []string{
+				iv,
+				val,
+				buildResult.Build.VariableRefs[iv].Type,
+				buildResult.Build.VariableRefs[iv].Source,
+			}
+			tbl.Rich(
+				columns,
+				[]string{
+					terminal.Green,
+				},
+			)
+		}
```

Comment:
Additionally, I'd love for the `Type` column here to call out when a variable is sensitive. The SHA is a good indicator, but being explicit will be very helpful too:

```
» Variables used:
     VARIABLE    |                     VALUE                      |  TYPE  | SOURCE
-----------------+------------------------------------------------+--------+----------
  registry_local | 7c9863a97af3acabf62da25c6e6f1b82483d635afad14b | sensitive,bool   | cli
                 | 0e5e445266420fa0d6                             |                  |

```

My formatting is off but you get the idea 😄

Reply:
iirc from when Rae & I talked about this, we don't want to explicitly say that it's a sensitive value to reduce risk.

Reply:
iirc from when Rae & I talked about this, we don't want to explicitly say that it's a sensitive value to reduce risk.

Reply:
> iirc from when Rae & I talked about this, we don't want to explicitly say that it's a sensitive value to reduce risk.

I think we should encode this information in the API (see my proto proposed changes). Terraform shows when a variable is sensitive or not. 

And I can't think of any threat model reason to not share this information. I know we don't have a threat model right now, but I can't think of anything hiding this information protects against. The protection itself is based on SHA-256 being safe, knowing it is a SHA-256 doesn't give you any head starts. In my opinion, it can also be trivially assumed to be sensitive when the value is obviously a SHA-256 hash and the value is non-sensical for the variable name.